modules/queue/workerpool.go

```diff
@@ -277,6 +291,22 @@ func (p *WorkerPool) addWorkers(ctx context.Context, cancel context.CancelFunc,
 				p.cond.Broadcast()
 				cancel()
 			}
+
+			select {
+			case <-p.baseCtx.Done():
+				// Don't warn if the baseCtx is shutdown
+			default:
+				if p.hasNoWorkerScaling() {
```

Comment:
Why does this differ from the other PR?

Reply:
because of the lack of the pause function in 1.16.

Reply:
I'm gonna send another pr against main simplifying the logic here as we can just test the baseCtx.Done() once and there's no point trying to boost if the queue is non-scaling.